plugins/inputs/procstat/filter.go

```diff
@@ -202,6 +202,7 @@ func (f *Filter) ApplyFilter() ([]processGroup, error) {
 					tags[k] = v
 				}
 				tags["parent_pid"] = strconv.FormatInt(int64(p.Pid), 10)
+				tags["child_level"] = strconv.FormatInt(int64(depth), 10)
```

Comment:
I think this should be a configuration option since it is a new tag

Reply:
I agree as it will also likely produce high cardinality. How about adding this as a field instead?

Reply:
I think the cardinality is higher on the parent_pid field so this change is less impactful than one might think.

Reply:
In any case, we cannot add a new tag unconditionally...

Reply:
Okay, I understand your point of view.
What option would you suggest to add this tag?
Is there a naming rule?
It also means that for each plugin each time new tags are added, an option is needed. Is this likely to create plugins with long configurations?
We could put:
tag_child_level.
If not defined, the tag is not added.
If tag_child_level=toto then the level is added in the tag "toto"

Reply:
up ? ;-)

Reply:
> I think this should be a configuration option since it is a new tag
@DStrand1 Have you a name recommanded for activate new tag ?

Reply:
> [...] It also means that for each plugin each time new tags are added, an option is needed. Is this likely to create plugins with long configurations? [...]

That's why we prefer "lists" of what to add instead of boolean options. This way we have one option with many potential values. For this plugin there actually already is the `tag_with` option, so why don't you just add the child-level there?
Check the code for how this option is used to conditionally add tags...

Reply:
Thanks.
I tryed, but I have 2 errors that I doesn't understand ...